src/hooks/useAsync.ts

```diff
@@ -62,17 +74,12 @@ const useAsync = (initialState?: any) => {
     );
 
     const run = useCallback(
-        (promise) => {
-            if (!promise || !promise.then) {
-                throw new Error(
-                    `The argument passed to useAsync().run must be a promise. Maybe a function that's passed isn't returning anything?`
-                );
-            }
+        (promise: Promise<T>) => {
             safeSetState({ status: States.LOADING });
             return promise.then(
-                (runResponseData: any) => {
-                    setData(runResponseData);
-                    return runResponseData;
+                (runResponse) => {
+                    setData(runResponse);
```

Comment:
It may be worthwhile to touch base on basic formatting standards for the code base as well. Code density tends to fluctuate on a file basis.

Reply:
I don't really care much about formatting unless it can be handled with a linter... so basically automatically handled. Otherwise I feel it is something that ends up being worth less than the amount of energy it takes to enforce.

So if there is formatting you'd like to see check if Prettier or Eslint can help and then we can decide.